src/business-logic/services/interfaces/inewsgateway-device-connection.ts

```diff
@@ -0,0 +1,15 @@
+import { INewsGatewayDeviceConnection } from '../inews-gateway-connection-implementation'
+
+export interface DeviceConnection {
+  connect(): Promise<boolean>
+  disconnect(): Promise<boolean>
+  send(deviceId: string, params: string[]): Promise<void>
+  listen(deviceId: string, callback: (data: unknown) => void): Promise<void>
```

Comment:
When I made the device connection service I did not know the nature of the conncections it would serve.

Would they be outbound only, or also inbound? 
Would they be all kinds of connections? TCP, UDP, web sockets, MQTT, CoAP, RESTful, AMQP, DDS, something else?

So I tried to make some assumptions based on similarities within these various protocols, but as you well know one size fits all seldom fit anything.

In the case of websocket comms with inewsgateway, I kind of want both, primarily want data events, but am unsure if DeviceConnection should handle everything so the service doesn't have to (in which case the callback wouldn't be needed at all).

Reply:
A very good point! :) The simplest course of action is probably to sculpt this when integrating to the new iNews Gateway, and try to incorporate it as a somewhat general interface.